Sources/Datadog/Core/Upload/DataUploadDelay.swift

```diff
@@ -15,12 +15,12 @@ internal struct DataUploadDelay {
 
     private var delay: TimeInterval
 
-    init(performance: PerformancePreset) {
-        self.defaultDelay = performance.defaultLogsUploadDelay
-        self.minDelay = performance.minLogsUploadDelay
-        self.maxDelay = performance.maxLogsUploadDelay
-        self.decreaseFactor = performance.logsUploadDelayDecreaseFactor
-        self.delay = performance.initialLogsUploadDelay
+    init(performance: UploadPerformancePreset) {
+        self.defaultDelay = performance.defaultUploadDelay
+        self.minDelay = performance.minUploadDelay
+        self.maxDelay = performance.maxUploadDelay
+        self.decreaseFactor = performance.uploadDelayDecreaseFactor
+        self.delay = performance.initialUploadDelay
```

Comment:
it seems like `DataUploadDelay` is made redundant as a duplicate of `UploadPerformancePreset`
they have the exact same properties with almost the same names.

could we remove it?

Reply:
`DataUploadDelay` is more than just a container for constants:
```swift
/// Mutable interval used for periodic data uploads.
internal struct DataUploadDelay {
   // ...

   mutating func nextUploadDelay() -> TimeInterval
   mutating func decrease()
   mutating func increaseOnce()
}
```

It calculates the next upload delay which changes within a ranges described by `UploadPerformancePreset`. It's a separate responsibility and I wouldn't mix it with `UploadPerformancePreset`.